src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc
             using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))
             {
                 var buffer = CreateTextBuffer(reader, cancellationToken);
-                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);
+                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);
```

Comment:
@Pilchie the `Encoding.Unicode` will insert a BOM by default. 

I think we should be using `Utf8Encoding` without the BOM